splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java

```diff
@@ -40,22 +39,19 @@
 
     private final AtomicReference<CurrentNetwork> currentNetwork = new AtomicReference<>();
 
-    ConnectionUtil(Context context) {
-        this(ConnectionUtil::createNetworkMonitoringRequest, NetworkDetector.create(context), (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE));
-    }
-
-    //for testing, since building the NetworkRequest fails in junit due to .. Android.
-    ConnectionUtil(Supplier<NetworkRequest> createNetworkMonitoringRequest, NetworkDetector networkDetector, ConnectivityManager connectivityManager) {
+    ConnectionUtil(NetworkDetector networkDetector) {
+        this.networkDetector = networkDetector;
         this.connectionMonitor = new ConnectionMonitor();
```

Comment:
This could be moved to the declaration.

Reply:
well... then we'd be leaking ourselves even *before* the constructor, since `ConnectionMonitor` is not a static class.

Reply:
Or you could move the `ConnectionStateListener` field to `ConnectionUtil` and not store `ConnectionMonitor` in a field at all - simply create a new instance in `startMonitoring()` and pass it to the `ConnectivityManager`. You'll avoid the dependency cycle between those two classes this way.